src/Diagnostics/CodeAnalysis/Core/MetaAnalyzers/RegisterActionAnalyzer.cs

```diff
@@ -49,22 +51,57 @@ public abstract class RegisterActionAnalyzer<TClassDeclarationSyntax, TInvocatio
             DiagnosticIds.InvalidSyntaxKindTypeArgumentRuleId,
             s_localizableTitleInvalidSyntaxKindTypeArgument,
             s_localizableMessageInvalidSyntaxKindTypeArgument,
-            "AnalyzerCorrectness",
+            DiagnosticCategory.AnalyzerCorrectness,
             DiagnosticSeverity.Warning,
             isEnabledByDefault: true,
             description: s_localizableDescriptionInvalidSyntaxKindTypeArgument,
             customTags: WellKnownDiagnosticTags.Telemetry);
 
+        private static readonly LocalizableString s_localizableDescriptionStatefulAnalyzerRegisterActionsDescription = new LocalizableResourceString(nameof(CodeAnalysisDiagnosticsResources.StatefulAnalyzerRegisterActionsDescription), CodeAnalysisDiagnosticsResources.ResourceManager, typeof(CodeAnalysisDiagnosticsResources), nameof(TLanguageKindEnumName));
```

Comment:
We've collectively fallen into a convention of putting the descriptors at the beginning of an analyzer, which is kind of weird given that it's stuff one rarely needs to read. I suggest a convention of putting the SupportedDiagnostics property first, then Initialize, then other action-related stuff, and the descriptors and strings at the bottom. (I'm not suggesting that you should necessarily do this for this change set, but that we might think about the convention.)